concepts/decorators/about.md

```diff
@@ -1,2 +1,246 @@
-#TODO: Add about for this concept.
-
+# About
+
+Decorators are functions that take another function as an argument for the purpose of extending or replacing the behavior of the passed-in function.
+If function `A` is a decorator, and function `B` is its argument, then function `A` modifies, extends or replaces function `B`'s **behavior** _without modifying_ function `B`'s code.
+We say that the decorator function `A` _wraps_ function `B`.
+While we talk about "modifying" behavior, the wrapped function is _not actually changed_.
+Behavior is either added _around_ the wrapped function (_and what it returns_), or the wrapped function's behavior is _substituted_ for some other behavior.
+
+## A Decorator is a Higher-Order Function
+
+A [higher-order function][higher-order functions] is a function that accepts one or more functions as arguments and/or returns one or more functions.
+A function, used as an argument or returned from another function, is a [first-class function][first-class functions].
+A Python function, as a [callable object][callable objects], is a first-class function which can be stored in a variable or used as an argument, much like any other value or object.
+Higher-order functions and first-class functions work together to make decorators possible.
+
+## What Using Decorator Looks Like
+
+The `@` symbol is prepended to the name of the decorator function and placed just above the function to be decorated, like so:
+
+```python
+@decorator
+def decorated_function():
+    pass
+```
+
+Some decorators accept arguments:
+
+```python
+@decorator_with_arg(name="Bob")
+def decorated_function2():
+    pass
+```
+
+If a decorator has defined default arguments, you must use parenthesis in the `@decorator()` call for the decorator to work:
```

Comment:
Why?  We might want to explain what the problem is being presented to the parser when there is no parenthesis.  Or pose it as a "Getting to Deep Dive Python" point…

Reply:
Next commit will include `, as you would in calling any function:`